sqlmesh/core/table_diff.py

```diff
@@ -83,14 +85,18 @@ def __init__(
         where: t.Optional[str | exp.Condition] = None,
         dialect: DialectType = None,
         limit: int = 20,
+        source_env: str = "source",
+        target_env: str = "target",
     ):
         self.adapter = adapter
         self.source = source
         self.target = target
         self.where = exp.condition(where, dialect=dialect) if where else None
         self.limit = limit
+        self.source_env = source_env
+        self.target_env = target_env
 
-        if isinstance(on, list):
+        if isinstance(on, (list, tuple)):
```

Comment:
FYI `click.option` with `multiple=True` returns a **tuple**, hence this addition. The `multiple=True` is a nice, call it idiomatic, way to support passing a param one or more times without ambiguity

Reply:
Without this, the CLI -- even rebased on latest main, was still fundamentally broken for me because `--on` was just a string in the `kwargs` and it would never hit this code path. There was no way for it to be a list without `multiple` or custom parsing.
cc: @tobymao